src/com/skillsup/dao/UserInFileDaoImpl.java

```diff
@@ -0,0 +1,77 @@
+package com.skillsup.dao;
+
+import com.skillsup.model.Gender;
+import com.skillsup.model.User;
+import com.skillsup.model.UserRole;
+
+import java.io.*;
+import java.math.BigDecimal;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+
+public class UserInFileDaoImpl implements UserDAO {
+    private static final String PATH_USER_DB_FILE = "E:\\onlineShop\\src\\com\\skillsup\\resources\\userDB.txt";
```

Comment:
its better to use relative path instead of absolute, "E:\\onlineShop\\src\\com\\skillsup\\resources\\userDB.txt" is not reusable across different PCs

Reply:
And You are absolutly right again! Shame on me!